src/widget/wstarrating.cpp

```diff
@@ -162,22 +118,16 @@ int WStarRating::starAtPosition(int x) {
 }
 
 void WStarRating::mouseReleaseEvent(QMouseEvent* /*unused*/) {
-    if (!m_pCurrentTrack) {
-        return;
-    }
-
-    m_pCurrentTrack->setRating(m_starRating.starCount());
+    emit ratingChanged(m_starRating.starCount());
 }
 
 void WStarRating::fillDebugTooltip(QStringList* debug) {
     WWidget::fillDebugTooltip(debug);
 
-    QString currentRating = "-";
+    QString currentRating;
+    currentRating.setNum(m_currRating);
     QString maximumRating = QString::number(m_starRating.maxStarCount());
 
-    if (m_pCurrentTrack) {
-        currentRating.setNum(m_pCurrentTrack->getRating());
-    }
```

Comment:
showing `- / 5` in the debug tooltip with no track would be the only reason to track the "track loaded" state. not worth the effort IMO